src/sentry.js

```diff
@@ -45,6 +51,82 @@ const preventNoise = (): void => {
   }
 };
 
+/**
+ * An HTTP request made by the app.
+ *
+ * One of the builtin "recognized breadcrumb types":
+ *   https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/#breadcrumb-types
+ */
+type HttpBreadcrumb = $ReadOnly<{|
+  ...Breadcrumb,
+  type: 'http',
+  data: {|
+    url?: string,
+    method?: string,
+    status_code?: number,
+    reason?: string,
+  |},
+|}>;
+
+/**
+ * Scrubs possibly personal information from a breadcrumb.
+ *
+ * - Removes the realm hostnames in API requests, except for Kandra-hosted
+ *   realms.
+ */
+function scrubBreadcrumb(breadcrumb: Breadcrumb, hint?: BreadcrumbHint): Breadcrumb {
+  switch (breadcrumb.type) {
+    case 'http': {
+      // $FlowIgnore[incompatible-indexer]
+      // $FlowIgnore[incompatible-type]
+      const httpBreadcrumb: HttpBreadcrumb = breadcrumb;
```

Comment:
Should have a comment explaining why we believe the ignores are OK.

Basically we're assuming that if `type` is `'http'`, then the breadcrumb has the shape of an `HttpBreadcrumb`. That assumption isn't expressed in the libdef, but is in the API docs at the page you link to.